src/ipc/ipc-helper.c

```diff
@@ -197,7 +198,7 @@ int comp_buffer_connect(struct comp_dev *comp, uint32_t comp_core,
 	return pipeline_connect(comp, buffer, dir);
 }
 
-int ipc_pipeline_complete(struct ipc *ipc, uint32_t comp_id)
+__cold int ipc_pipeline_complete(struct ipc *ipc, uint32_t comp_id)
```

Comment:
Can you check this, not sure if done in prepare() ? maybe for IPC3 only ?

Reply:
@lgirdwood this is only called from
```
ipc4_pipeline_complete()
ipc4_pipeline_prepare()
ipc4_set_pipeline_state()				idc_ppl_state()
ipc4_process_glb_message()				idc_cmd()
ipc_cmd()						idc_handler()
ipc_platform_do_cmd()		idc_ipc()		P4WQ
ipc_do_cmd()			idc_cmd()
EDF scheduler			idc_handler()
				P4WQ
```
so, it's only called from the EDF scheduler or from IDC P4WQ, both of which use the `EDF_ZEPHYR_PRIORITY` priority (currently 1)

Reply:
ok, so are you confirming its not called as part of a LL or DP process() ? I assume its only used in EDF for non process() usage.

Reply:
@lgirdwood correct, as it stands there don't seem to be any paths leading to this function being called from audio processing context